### PR TITLE
Extract PBI dataset parameters [sc-25094]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

Some power query expression needs parameters to derive the physical dataset

### 🤓 What?

- add query to fetch PowerBI dataset parameters
- The compiled model has changed TypeEnum to LogType

### 🧪 Tested?

unit test
Tested against metaphor and customer environment

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/power_bi/power_bi_client.py [hunk 1]
```diff
@@ -128,6 +128,24 @@ def get_datasets(self, group_id: str) -> List[PowerBIDataset]:
...
+
+        try:
+            return self._call_get(
+                url,
+                List[PowerBIDatasetParameter],
+                transform_response=lambda r: r.json()["value"],
+            )
+        except Exception:
+            logger.exception(
+                f"Unable to get parameters for dataset {dataset_id} in group {group_id}"
```
Comment (mars-lan) [inline metaphor/power_bi/power_bi_client.py:145]: Also add the following message:
```
Please add the service principal as a viewer to the workspace
```

File: pyproject.toml [hunk 2]
```diff
@@ -19,7 +19,7 @@ asyncpg = { version = "^0.29.0", optional = true }
 avro = { version = "^1.11.3", optional = true }
 aws-assume-role-lib = "^2.10.0"
 azure-identity = { version = "^1.14.0", optional = true }
-azure-mgmt-datafactory = { version = "^3.1.0", optional = true }
+azure-mgmt-datafactory = { version = "^6.0.0", optional = true }
```
Comment (mars-lan) [inline pyproject.toml:22]: No breaking changes?

File: metaphor/power_bi/power_bi_client.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/power_bi/power_bi_client.py]: updated

File: pyproject.toml (hunk 2 above)
Comment (alyiwang) [reply pyproject.toml:22]: tested the azure_data_factory crawler locally.

**Review by mars-lan — APPROVED** [review]

**Review by mars-lan — APPROVED** [review]

**Pull request merged**